contrib/p4/p4execution.py

```diff
@@ -0,0 +1,838 @@
+# SPDX-FileCopyrightText: 2020-2021 CASTOR Software Research Centre
+# <https://www.castor.kth.se/>
+# SPDX-FileCopyrightText: 2020-2021 Johan Paulsson
+
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import os
+import logging
+
+import time
+import re
+import threading
+from benchexec import systeminfo
+from benchexec.model import Run
+from p4.p4_run_setup import P4SetupHandler
+from p4.Counter import Counter
+
+from benchexec import tooladapter
+from benchexec import util
+from benchexec import benchexec
+from benchexec import BenchExecException
+
+# File handling
+import zipfile
+from shutil import copyfile, rmtree
+import json
+from distutils.dir_util import copy_tree
+
+try:
+    import docker
+except Exception as e:
```

Comment:
```suggestion
except ImportError:
```
It is always a good idea to make exception catching as specific as possible (and rarely or never catch `Exception`).